### PR TITLE
Add support for 'WHERE x IN UNNEST(...)'

### PR DESCRIPTION
**Pull request by sid-kap**
The BigQuery [`IN` operator](https://cloud.google.com/bigquery/docs/reference/standard-sql/operators#in_operators) supports `UNNEST` arguments:
```
search_value [NOT] IN value_set

value_set:
  {
    (expression[, ...])
    | (subquery)
    | UNNEST(array_expression)
  }
```

This adds support for parsing queries with `WHERE x IN UNNEST(...)` clauses.

### CODE REVIEW

**Comment by sid-kap** [comment]
It seems like MySQL and Postgres don't support this. Should I set up the parser to only allow this if parsing BigQuery SQL?

**Comment by tobymao** [comment]
in general parsers are more lenient. 

sqlglot should parse any sql, but never generate bad sql

**Review by tobymao — APPROVED** [review]

**Comment by sid-kap** [comment]
Hmm. In that case, we should probably make generator.py fail when rendering this in other dialects (or use a workaround like `SELECT UNNEST(...)`)

**Comment by tobymao** [comment]
can you convert IN unnest() into something that works in all dialects?

if not, you can override other dialects or log unsupported_error

**Comment by tobymao** [comment]
i will say though that this is not a requirement to merge this PR in as in unnest is probably pretty rare. but if you want to solve it go for it

**Comment by sid-kap** [comment]
It looks like `SELECT UNNEST` works in postgres! I'll update the PR in the evening

File: sqlglot/generator.py [hunk 1]
```diff
@@ -798,13 +798,18 @@ def if_sql(self, expression):
...
+        if query:
+            in_sql = self.wrap(query)
+        elif unnest:
+            in_sql = self.in_unnest_op(unnest)
+        else:
+            in_sql = f"({self.expressions(expression, flat=True)})"
         return f"{self.sql(expression, 'this')} IN {in_sql}"
 
+    def in_unnest_op(self, unnest):
+        return f"(SELECT {self.unnest_sql(unnest)})"
```
Comment (tobymao) [inline sqlglot/generator.py:811]: self.sql(unnest)

File: sqlglot/dialects/bigquery.py [hunk 2]
```diff
@@ -92,6 +92,9 @@ class Generator(Generator):
             exp.DataType.Type.TEXT: "STRING",
         }
 
+        def in_unnest_op(self, unnest):
+            return self.unnest_sql(unnest)
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:96]: self.sql(unnest)

**Comment by sid-kap** [comment]
@tobymao Thanks for reviewing/merging these so quickly!
Do you know when the next release is scheduled for?

**Comment by tobymao** [comment]
i need to plan that out, the current main branch has many breaking changes so i may bump to v5

**Comment by tobymao** [comment]
will probably release v5 tomorrow

**Comment by sid-kap** [comment]
Great, thanks so much!

**Pull request merged**